doc/rst/technotes/operatorMethods.rst

```diff
@@ -1,129 +0,0 @@
-.. _readme-operator-methods:
-
-.. default-domain:: chpl
-
-================
-Operator Methods
-================
-
-This README describes support in the Chapel compiler for defining operators
-as methods on types.  These features are still in the process of being improved.
-This README is intended to cover the expected behavior, but the expected
-behavior may change as experience with the feature is gained.
-
-Defining Operators
-------------------
-
-Operators may be overloaded to support new behavior on one or more types using
-the ``operator`` keyword.  Such overloads may be defined as standalone
-functions, e.g.
-
-.. code-block:: chapel
-
-   operator +(lhs: t1, rhs: t2) { ... }
-
-or as methods defined on a particular type, e.g.
-
-.. code-block:: chapel
-
-   record R {
-     var intField: int;
-
-     operator +(lhs: R, rhs: R) {
-       return new R(lhs.intField + rhs.intField);
-     }
-   }
-
-This README will focus on operator methods.
-
-Operator methods are equivalent to type methods - the type on which the operator
-is declared causes the operator to have method-like visibility for that type,
-and the ``this`` receiver is a ``type``.
-
-Operator methods may be defined as primary, secondary, or tertiary methods.  For
-example, the following code defines a primary ``+`` operator and a secondary
-``-`` operator on a record:
-
-.. code-block:: chapel
-
-   record R {
-     var intField: int;
-
-     operator +(lhs: R, rhs: R) {
-       return new R(lhs.intField + rhs.intField);
-     }
-   }
-
-   operator R.-(lhs: R, rhs: R) {
-     return new R(lhs.intField - rhs.intField);
-   }
-
-
-The method receiver for an operator method will be used to determine when that
-operator is visible.  This behavior is most useful when the method receiver
-matches the type of at least one of the other arguments to the operator.
-However, it is possible to define an operator method where the receiver type
-does not match the type for any other argument.
-
-Operator methods can be defined on concrete types, generic types, or particular
-instantiations of generic types.
-
-Operator Methods and Classes
-----------------------------
-
-Operators that are defined on classes that are inherited from can run into known
-issues. To avoid these issues, define the class arguments to operator methods
-with ``borrowed``, e.g.:
-
-.. code-block:: chapel
-
-   class Parent {
-     var parentField: int;
-
-     operator <(lhs: borrowed Parent, rhs: borrowed Parent) {
-       return lhs.x < rhs.x;
-     }
-   }
-
-   class Child: Parent {
-     var childField: bool;
-   }
```

Comment:
This section doesn't have an equivalent in the spec version. Is it covered elsewhere? Did you remove it on purpose?

Reply:
Yeah, sorry not to have mentioned it!  This section describes a workaround, but the spec is intended as an aspirational document so it didn't really seem appropriate to include it there.  This section actually made me hesitant to move the document over, but we've lived with operator methods for a while so it really seems like it was time for it to be in the spec.  We've already got a future and issue for it not working, including the code for how to avoid it.

Should I call out dropping the section in the PR message?  Do something to include it?

Reply:
It'd be great to mention it in the PR message and link to that issue.

As far as the spec goes, I think it's OK to leave this out. It'd be nice if we could get a compiler warning for operators on classes that don't use `borrowed` for now, assuming the issues still exist. But all of that can be future work & covered by the issue.